pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -178,10 +180,13 @@ func (configgen *ConfigGeneratorImpl) buildClusters(proxy *model.Proxy, req *mod
 		// Setup inbound clusters
 		inboundPatcher := clusterPatcher{efw: envoyFilterPatches, pctx: networking.EnvoyFilter_SIDECAR_INBOUND}
 		clusters = append(clusters, configgen.buildInboundClusters(cb, proxy, instances, inboundPatcher)...)
-		clusters = append(clusters, configgen.buildInboundHBONEClusters(cb, proxy, instances)...)
 		// Pass through clusters for inbound traffic. These cluster bind loopback-ish src address to access node local service.
 		clusters = inboundPatcher.conditionallyAppend(clusters, nil, cb.buildInboundPassthroughClusters()...)
 		clusters = append(clusters, inboundPatcher.insertedClusters()...)
+		if proxy.EnableHBONE() {
+			clusters = append(clusters, outboundTunnelCluster(proxy, req.Push))
```

Comment:
@howardjohn I guess, we do not want envoy filter patches to be applied to these clusters - right/

Reply:
Seems reasonable, we don't have proper matching API I would think

Reply:
We need it for gateway though

Reply:
Ah. I thought it is only for Sidecar. I will make that change

Reply:
Why should we distinguish cluster for envoyfilter patch? Maybe tunnel cluster need to be mutated for some specific usage in the future

Reply:
For now we don't have a way to match the cluster though. It doesn't have a port or destination or anything like that?

Reply:
Even not possible via cluster name? I remember we can match by cluster name

Reply:
it is possible. But IMO we should wait till we get through tests and some level of usage before we enable envoy filters for this.